README.md

```diff
@@ -137,6 +137,71 @@ You can extend your data-types to work with `matcher-combinators` by implemented
 
 An example of this in the wild can be seen in the `abracad` library [here](https://github.com/nubank/abracad/blob/b52e6a7114461f50bdacc2cf09a1de08f707b9f3/test/abracad/custom_types_test.clj#L15-L20).
 
+## Overriding default matchers
```

Comment:
these docs aren't the greatest but I think the whole readme needs to be redone so I'm not going to refine it right now